lib/widgets/app.dart

```diff
@@ -25,10 +26,29 @@ class ZulipApp extends StatelessWidget {
       //   https://m3.material.io/theme-builder#/custom
       colorScheme: ColorScheme.fromSeed(seedColor: kZulipBrandColor));
     return GlobalStoreWidget(
-      child: MaterialApp(
-        title: 'Zulip',
-        theme: theme,
-        home: const ChooseAccountPage()));
+      child: BrowserLoginWidget(
+        child: Builder(
+          builder: (context) => MaterialApp(
+            title: 'Zulip',
+            theme: theme,
+            home: const ChooseAccountPage(),
+            navigatorKey: BrowserLoginWidget.of(context).navigatorKey,
+            // TODO: Migrate to `MaterialApp.router` & `Router`, so that we can receive
+            //       a full Uri instead of just path+query components and also maybe
```

Comment:
Hmm, I was going to write that I believe using `Router` wouldn't change this, because if you trace through how the information flows through the engine and into the Flutter framework code, you find that the Dart side only ever sees path, query, and fragment.

… But it turns out that's because my engine tree was a few months old! This feature was added recently in https://github.com/flutter/flutter/issues/100624 . Now Dart does see the full URL.

Still, I think we can do this orthogonally to any switch to `Router`. The key is to register a `WidgetsBindingObserver` and implement `didPushRouteInformation`:
https://api.flutter.dev/flutter/widgets/WidgetsBindingObserver/didPushRouteInformation.html

I think that can then replace the job that this `onGenerateRoute` is doing.